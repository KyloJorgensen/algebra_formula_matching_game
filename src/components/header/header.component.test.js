import React from 'react';
import { shallow } from 'enzyme';
import Header from './header.component.js';

describe('Render Header', () => {
	let wrapper;
	beforeEach(() => {
		wrapper = shallow(<Header />);
	});

	it('should match the snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('should have a button', () => {
		let button = wrapper.find('button');
		expect(button).toHaveProperty('length', 1);
	});

	it('should have a input', () => {
		let input = wrapper.find('input');
		expect(input).toHaveProperty('length', 1);
	});

	it('should have newGame as a default function that is returns nothing', () => {
		expect(Header.defaultProps.newGame()).toBeUndefined();
	});

	it('should have onChange as a default function that is returns nothing', () => {
		expect(Header.defaultProps.onChange()).toBeUndefined();
	});
});

describe('interaction', () => {
	let props;
	let wrapper;
	beforeEach(() => {
		props = {
			newGame: () => 1,
			onChange: () => 1,
			matches: 12,
		};
	});
	describe('New Game Button', () => {
		beforeEach(() => {
			props.newGame = jest.fn().mockImplementation();
			wrapper = shallow(<Header {...props} />)
		});
		describe('new Game button', () => {
			let button;
			beforeEach(() => {
				button = wrapper.find('button');
				button.simulate('click');
			})
			it('when button is click run function', () => {
				const mockCallLength = props.newGame.mock.calls.length;
				expect(mockCallLength).toEqual(1);
			});
		});
	});
	describe('Total Matches Input', () => {
		beforeEach(() => {
			props.onChange = jest.fn().mockImplementation();
			props.matches = 23;
			wrapper = shallow(<Header {...props} />)
		});

		it('should be 23', () => {
			const input = wrapper.find('input')
			const matches = input.props().value
			expect(matches).toEqual(23);
		});

		it('onChange should be called on change', () => {
			const input = wrapper.find('input');
			input.simulate('change', { target: { value: 17 } });
			const mockCallLength = props.onChange.mock.calls.length;
			expect(mockCallLength).toEqual(1);
		});

		it('value should be chagned to 17', () => {
			const input = wrapper.find('input');
			input.simulate('change', { target: { value: 17 } });
			const mockCallResult = props.onChange.mock.calls[0][0];
			expect(mockCallResult.target.value).toEqual(17);
		});


	});
});
