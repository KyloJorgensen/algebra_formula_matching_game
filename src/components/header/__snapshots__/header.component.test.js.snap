// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Header should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    matches={12}
    newGame={[Function]}
    onChange={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Algebra Formula Matching Game
        </h1>,
        <div
          className="total-matches-wrapper"
        >
          <div>
            <label
              htmlFor="total-matches"
            >
              Total Matches
            </label>
            <input
              id="total-matches"
              onChange={[Function]}
              type="number"
              value={12}
            />
          </div>
        </div>,
        <button
          className="new-game"
          onClick={[Function]}
        >
          New Game
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Algebra Formula Matching Game",
        },
        "ref": null,
        "rendered": "Algebra Formula Matching Game",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            <label
              htmlFor="total-matches"
            >
              Total Matches
            </label>
            <input
              id="total-matches"
              onChange={[Function]}
              type="number"
              value={12}
            />
          </div>,
          "className": "total-matches-wrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                htmlFor="total-matches"
              >
                Total Matches
              </label>,
              <input
                id="total-matches"
                onChange={[Function]}
                type="number"
                value={12}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Total Matches",
                "htmlFor": "total-matches",
              },
              "ref": null,
              "rendered": "Total Matches",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "total-matches",
                "onChange": [Function],
                "type": "number",
                "value": 12,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "New Game",
          "className": "new-game",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "New Game",
        "type": "button",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Algebra Formula Matching Game
          </h1>,
          <div
            className="total-matches-wrapper"
          >
            <div>
              <label
                htmlFor="total-matches"
              >
                Total Matches
              </label>
              <input
                id="total-matches"
                onChange={[Function]}
                type="number"
                value={12}
              />
            </div>
          </div>,
          <button
            className="new-game"
            onClick={[Function]}
          >
            New Game
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Algebra Formula Matching Game",
          },
          "ref": null,
          "rendered": "Algebra Formula Matching Game",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div>
              <label
                htmlFor="total-matches"
              >
                Total Matches
              </label>
              <input
                id="total-matches"
                onChange={[Function]}
                type="number"
                value={12}
              />
            </div>,
            "className": "total-matches-wrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label
                  htmlFor="total-matches"
                >
                  Total Matches
                </label>,
                <input
                  id="total-matches"
                  onChange={[Function]}
                  type="number"
                  value={12}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Total Matches",
                  "htmlFor": "total-matches",
                },
                "ref": null,
                "rendered": "Total Matches",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "total-matches",
                  "onChange": [Function],
                  "type": "number",
                  "value": 12,
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "New Game",
            "className": "new-game",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "New Game",
          "type": "button",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
